src/services/tests/gatewayClient.test.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  status_success_response,
+  settings,
+  search_success_response,
+  index_success_response,
+} from "./gatewayClient.testData"
+
+import AxiosMockAdapter from "axios-mock-adapter"
+import gatewayClient, { gatewayInstance } from "../gatewayClient"
+import logger from "../../logger"
+
+let mockGatewayClient: AxiosMockAdapter
+const mockConnectionCallback = jest.fn()
+
+describe("when connecting to gateway", () => {
+  beforeAll(() => {
+    mockGatewayClient = new AxiosMockAdapter(gatewayInstance)
+  })
+
+  beforeEach(() => {
+    mockGatewayClient.reset()
+  })
+
+  it("should trigger a success messsage when connceted successfully", async () => {
+    mockGatewayClient.onGet("/status").reply(200, status_success_response)
+
+    await gatewayClient.connect(settings, mockConnectionCallback)
+    expect(mockConnectionCallback.mock.calls[0][0]).toEqual({
+      connected: true,
+      message: `Successfully connected to Jina at ${settings.gatewayHost}:${settings.gatewayPort}`,
+    })
+  })
+
+  it("should trigger a success log when searched successfully", async () => {
+    mockGatewayClient.onPost("api/search").reply(200, search_success_response)
+    const loggerSpy = jest.spyOn(logger, "log")
+    await gatewayClient.search("Josef Stalin")
```

Comment:
a bit of a wierd thing but maybe we should stay away from having any sorts of political context in here? just in case id replace it with "smiling unicorn" or whatever other non political entity

Reply:
changed it to smiling unicorn